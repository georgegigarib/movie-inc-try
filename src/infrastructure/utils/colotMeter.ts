export const getColorForVote = (measure: number): string => {
  if (measure == 0) {
    return '#CC000000';
  } else if (measure <= 1) {
    return '#CC0000';
  } else if (measure <= 2) {
    return '#CC3700';
  } else if (measure <= 3) {
    return '#CC513A';
  } else if (measure <= 4) {
    return '#CC7300';
  } else if (measure <= 5) {
    return '#CC8400';
  } else if (measure <= 6) {
    return '#CCAC00';
  } else if (measure <= 7) {
    return '#89CC0D';
  } else if (measure <= 8) {
    return '#65A800';
  } else if (measure <= 9) {
    return '#289B1A';
  } else {
    return '#004B00';
  }
};
