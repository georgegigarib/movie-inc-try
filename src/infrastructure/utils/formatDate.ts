export function formatDate(dateString: string): string {
    const [year, month, day] = dateString.split('-').map(Number);

    const months = [
        'Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun',
        'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec'
    ];

    return `${months[month - 1]} ${day}, ${year}`;
}


export function getYearFromDate (dateString: string): string {
    const yearMatch = dateString.match(/\d{4}$/);
    return yearMatch ? yearMatch[0] : 'Unknown';
  };