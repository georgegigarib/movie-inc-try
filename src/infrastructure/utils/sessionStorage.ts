import { GuestSession } from '@/src/domain/Guest/model/GuestSession';
import AsyncStorage from '@react-native-async-storage/async-storage';

const storageKey = 'MovieDBGuestSession';

export const getSession = async (): Promise<GuestSession | null> => {
  try {
      const data = await AsyncStorage.getItem(storageKey);
      return data ? JSON.parse(data) : null;
  } catch {
      return null;
  }
};

export const setSession = async (sessionData: GuestSession) => {
    try {
    await AsyncStorage.setItem(storageKey, JSON.stringify(sessionData));
    } catch {}
};

export const isActive = (session: GuestSession): boolean => {
  const formattedDateStr = session.expireDate.replace(' ', 'T').replace(' UTC', 'Z');

  return new Date().getTime() < new Date(formattedDateStr).getTime();
};