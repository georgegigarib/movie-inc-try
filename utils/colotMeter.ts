export const getColorForVote = (measure: number): string => {
  if (measure <= 1) {
    return '#CC0000'; // Rojo oscuro
  } else if (measure <= 2) {
    return '#CC3700'; // Naranja rojizo oscuro
  } else if (measure <= 3) {
    return '#CC513A'; // Rojo tomate oscuro
  } else if (measure <= 4) {
    return '#CC7300'; // Naranja oscuro
  } else if (measure <= 5) {
    return '#CC8400'; // Ámbar oscuro
  } else if (measure <= 6) {
    return '#CCAC00'; // Dorado oscuro
  } else if (measure <= 7) {
    return '#89CC0D'; // Verde amarillo oscuro
  } else if (measure <= 8) {
    return '#65A800'; // Verde oscuro
  } else if (measure <= 9) {
    return '#289B1A'; // Lima oscuro
  } else {
    return '#004B00'; // Verde oscuro profundo
  }
};
