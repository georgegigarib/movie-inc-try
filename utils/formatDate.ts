export function formatDate(dateString: string): string {
    const [year, month, day] = dateString.split('-').map(Number);

    const months = [
        'Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun',
        'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec'
    ];

    return `${months[month - 1]} ${day}, ${year}`;
}
