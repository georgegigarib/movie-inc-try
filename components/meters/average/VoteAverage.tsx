import { ThemedText } from "@/components/ThemedText";
import { getColorForVote } from "@/utils/colotMeter";
import React, { useEffect, useRef } from "react";
import { View, Text, StyleSheet, Animated } from "react-native";

interface VoteAverageProps {
  voteAverage: number | undefined;
}

export default function VoteAverage({ voteAverage }: VoteAverageProps) {
  const animatedColor = useRef(new Animated.Value(0)).current;
  const animatedValue1 = useRef(new Animated.Value(0)).current;
  const animatedValue2 = useRef(new Animated.Value(0)).current;
  const animatedValue3 = useRef(new Animated.Value(0)).current;

  useEffect(() => {
    Animated.timing(animatedColor, {
      toValue: 100,
      duration: 1000,
      useNativeDriver: false,
    }).start();

    const animateConditionally = (
      value: Animated.Value,
      conditions: { check: boolean; toValue: number }[]
    ) => {
      const condition = conditions.find((c) => c.check);
      if (condition) {
        Animated.timing(value, {
          toValue: condition.toValue,
          duration: 1000,
          useNativeDriver: false,
        }).start();
      }
    };

    if (voteAverage !== undefined) {
      const percentage = (voteAverage / 10) * 100;

      // AnimatedValue1
      animateConditionally(animatedValue1, [
        { check: voteAverage > 2.5 && voteAverage <= 5, toValue: percentage },
        { check: voteAverage < 2.5, toValue: 25 },
        { check: voteAverage > 5, toValue: 50 },
      ]);

      // AnimatedValue2
      animateConditionally(animatedValue2, [
        { check: voteAverage > 5 && voteAverage <= 7.5, toValue: percentage },
        { check: voteAverage < 5, toValue: 25 },
        { check: voteAverage > 7.5, toValue: 75 },
      ]);

      // AanimatedValue3
      animateConditionally(animatedValue3, [
        { check: voteAverage > 7.5 && voteAverage <= 10, toValue: percentage },
        { check: voteAverage < 7.5, toValue: 25 },
        { check: voteAverage > 10, toValue: 100 },
      ]);
    }
  }, [voteAverage]);

  const colorInterpolation = animatedColor.interpolate({
    inputRange: [0, 100],
    outputRange: [getColorForVote(0), getColorForVote(voteAverage!)],
  });

  const animatedValues = [animatedValue1, animatedValue2, animatedValue3];
  const rotations = animatedValues.map((animatedValue) =>
    animatedValue.interpolate({
      inputRange: [0, 100],
      outputRange: ["40deg", "400deg"],
    })
  );

  return (
    <View style={styles.container}>
      <View style={[styles.circle, { borderColor: "transparent" }]}>
        <Animated.View
          style={[
            styles.progressStatic,
            {
              borderColor: "transparent",
              borderLeftColor: colorInterpolation,
            },
          ]}
        />
        {rotations.map((rotation, index) => (
          <Animated.View
            key={index}
            style={[
              styles.progressMoving,
              {
                transform: [{ rotate: rotation }],
                borderColor: "transparent",
                borderLeftColor: colorInterpolation,
              },
            ]}
          />
        ))}
        <View style={styles.innerCircle}>
          <ThemedText style={styles.percentage}>
            {voteAverage !== undefined ? `${voteAverage.toFixed(1)}` : "N/A"}
          </ThemedText>
        </View>
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
  },
  circle: {
    width: 63,
    height: 63,
    borderRadius: 80,
    borderWidth: 1,
    justifyContent: "center",
    alignItems: "center",
    position: "relative",
    backgroundColor: "transparent",
  },
  progressStatic: {
    position: "absolute",
    width: "100%",
    height: "100%",
    borderRadius: 30,
    borderWidth: 6,
    borderColor: "transparent",
    transform: [{ rotate: "130deg" }],
  },
  progressMoving: {
    position: "absolute",
    width: "100%",
    height: "100%",
    borderRadius: 30,
    borderWidth: 6,
    borderColor: "transparent",
    transform: [{ rotate: "130deg" }],
  },
  innerCircle: {
    width: 54,
    height: 54,
    borderRadius: 45,
    backgroundColor: "white",
    justifyContent: "center",
    alignItems: "center",
  },
  percentage: {
    fontSize: 16,
    fontWeight: "bold",
  },
});
