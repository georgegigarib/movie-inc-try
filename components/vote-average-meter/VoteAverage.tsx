import { ThemedText } from "@/components/ThemedText";
import { ThemedView } from "@/components/ThemedView";
import { getColorForVote } from "@/utils/colotMeter";
import React, { useEffect, useRef } from "react";
import { View, StyleSheet, Animated } from "react-native";

interface VoteAverageProps {
  voteAverage: number | undefined;
}

export default function VoteAverage({ voteAverage }: VoteAverageProps) {
  const animatedColor = useRef(new Animated.Value(0)).current;
  const animatedValue1 = useRef(new Animated.Value(0)).current;
  const animatedValue2 = useRef(new Animated.Value(0)).current;
  const animatedValue3 = useRef(new Animated.Value(0)).current;

  const fixedAverage = (rate: number) => {
    return Math.floor(rate * 10) / 10
  }
  
  useEffect(() => {
    Animated.timing(animatedColor, {
      toValue: 100,
      duration: 1000,
      useNativeDriver: false,
    }).start();

    const animateConditionally = (
      value: Animated.Value,
      conditions: { check: boolean; toValue: number }[]
    ) => {
      const condition = conditions.find((c) => c.check);
      if (condition) {
        Animated.timing(value, {
          toValue: condition.toValue,
          duration: 1000,
          useNativeDriver: false,
        }).start();
      }
    };

    if (voteAverage !== undefined) {
      const percentage = (voteAverage / 10) * 100;

      animateConditionally(animatedValue1, [
        { check: voteAverage > 2.5 && voteAverage <= 5, toValue: percentage },
        { check: voteAverage < 2.5, toValue: 25 },
        { check: voteAverage > 5, toValue: 50 },
      ]);

      animateConditionally(animatedValue2, [
        { check: voteAverage > 5 && voteAverage <= 7.5, toValue: percentage },
        { check: voteAverage < 5, toValue: 25 },
        { check: voteAverage > 7.5, toValue: 75 },
      ]);

      animateConditionally(animatedValue3, [
        { check: voteAverage > 7.5 && voteAverage <= 10, toValue: percentage },
        { check: voteAverage < 7.5, toValue: 25 },
        { check: voteAverage > 10, toValue: 100 },
      ]);
    }
  }, [voteAverage]);

  const colorInterpolation = animatedColor.interpolate({
    inputRange: [0, 100],
    outputRange: [getColorForVote(0), getColorForVote(voteAverage!)],
  });

  const animatedValues = [animatedValue1, animatedValue2,];
  const rotations = animatedValues.map((animatedValue) =>
    animatedValue.interpolate({
      inputRange: [0, 100],
      outputRange: ["40deg", "400deg"],
    })
  );

  return (
    <View style={styles.container}>
      <View style={[styles.circle, { borderColor: "transparent" }]}>
        <Animated.View
          style={[
            styles.progressStatic,
            {
              borderColor: "transparent",
              borderLeftColor: colorInterpolation,
            },
          ]}
        />
        {rotations.map((rotation, index) => (
          <Animated.View
            key={index}
            style={[
              styles.progressMoving,
              {
                transform: [{ rotate: rotation }],
                borderColor: "transparent",
                borderLeftColor: colorInterpolation,
              },
            ]}
          />
        ))}
        <ThemedView style={styles.innerCircle}>
          <ThemedText style={styles.percentage}>
            {voteAverage !== undefined ? `${fixedAverage(voteAverage)}` : "N/A"}
          </ThemedText>
        </ThemedView>
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
  },
  circle: {
    width: 63,
    height: 63,
    borderRadius: 80,
    borderWidth: 0,
    justifyContent: "center",
    alignItems: "center",
    position: "relative",
    backgroundColor: "#D1D1D182",
  },
  progressStatic: {
    position: "absolute",
    width: "100%",
    height: "100%",
    borderRadius: 30,
    borderWidth: 6,
    borderColor: "transparent",
    transform: [{ rotate: "130deg" }],
  },
  progressMoving: {
    position: "absolute",
    width: "100%",
    height: "100%",
    borderRadius: 30,
    borderWidth: 6,
    borderColor: "transparent",
    transform: [{ rotate: "130deg" }],
  },
  innerCircle: {
    width: 54,
    height: 54,
    borderRadius: 45,
    justifyContent: "center",
    alignItems: "center",
  },
  percentage: {
    fontSize: 16,
    fontWeight: "bold",
  },
});
