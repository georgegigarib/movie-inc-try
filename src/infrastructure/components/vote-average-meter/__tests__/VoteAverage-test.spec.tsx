import React from "react";
import { render } from "@testing-library/react-native";
import VoteAverage from "@/src/infrastructure/components/vote-average-meter/VoteAverage";
import { getColorForVote } from "@/src/infrastructure/utils/colotMeter";

jest.mock("@/src/infrastructure/utils/colotMeter", () => ({
  getColorForVote: jest.fn().mockImplementation((vote) => `color-for-vote-${vote}`),
}));

describe("VoteAverage", () => {
  test.each([
    [1, 'color-for-vote-1'],
    [5, 'color-for-vote-5'],
    [8, 'color-for-vote-8'],
    [10, 'color-for-vote-10']
  ])(
    "renders correctly for voteAverage = %i",
    (voteAverage, expectedColor) => {
      const { toJSON } = render(<VoteAverage voteAverage={voteAverage} />);
      expect(toJSON()).toMatchSnapshot();

      expect(getColorForVote).toHaveBeenCalledWith(voteAverage);
    }
  );
});
