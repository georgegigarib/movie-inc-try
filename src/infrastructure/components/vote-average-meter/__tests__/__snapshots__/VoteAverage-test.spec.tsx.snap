// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VoteAverage renders correctly for voteAverage = 1 1`] = `
<View
  style={{}}
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "#D1D1D182",
          "borderRadius": 80,
          "borderWidth": 0,
          "height": 63,
          "justifyContent": "center",
          "position": "relative",
          "width": 63,
        },
        {
          "borderColor": "transparent",
        },
      ]
    }
  >
    <View
      collapsable={false}
      style={
        {
          "borderColor": "transparent",
          "borderLeftColor": "color-for-vote0",
          "borderRadius": 30,
          "borderWidth": 6,
          "height": "100%",
          "position": "absolute",
          "transform": [
            {
              "rotate": "130deg",
            },
          ],
          "width": "100%",
        }
      }
    />
    <View
      collapsable={false}
      style={
        {
          "borderColor": "transparent",
          "borderLeftColor": "color-for-vote0",
          "borderRadius": 30,
          "borderWidth": 6,
          "height": "100%",
          "position": "absolute",
          "transform": [
            {
              "rotate": "40deg",
            },
          ],
          "width": "100%",
        }
      }
    />
    <View
      collapsable={false}
      style={
        {
          "borderColor": "transparent",
          "borderLeftColor": "color-for-vote0",
          "borderRadius": 30,
          "borderWidth": 6,
          "height": "100%",
          "position": "absolute",
          "transform": [
            {
              "rotate": "40deg",
            },
          ],
          "width": "100%",
        }
      }
    />
    <View
      collapsable={false}
      style={
        {
          "borderColor": "transparent",
          "borderLeftColor": "color-for-vote0",
          "borderRadius": 30,
          "borderWidth": 6,
          "height": "100%",
          "position": "absolute",
          "transform": [
            {
              "rotate": "40deg",
            },
          ],
          "width": "100%",
        }
      }
    />
    <View
      style={
        [
          {
            "backgroundColor": "#fff",
          },
          {
            "alignItems": "center",
            "borderRadius": 45,
            "height": 54,
            "justifyContent": "center",
            "width": 54,
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#11181C",
            },
            {
              "fontSize": 16,
            },
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            {
              "fontSize": 16,
              "fontWeight": "bold",
            },
          ]
        }
      >
        1
      </Text>
    </View>
  </View>
</View>
`;

exports[`VoteAverage renders correctly for voteAverage = 5 1`] = `
<View
  style={{}}
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "#D1D1D182",
          "borderRadius": 80,
          "borderWidth": 0,
          "height": 63,
          "justifyContent": "center",
          "position": "relative",
          "width": 63,
        },
        {
          "borderColor": "transparent",
        },
      ]
    }
  >
    <View
      collapsable={false}
      style={
        {
          "borderColor": "transparent",
          "borderLeftColor": "color-for-vote0",
          "borderRadius": 30,
          "borderWidth": 6,
          "height": "100%",
          "position": "absolute",
          "transform": [
            {
              "rotate": "130deg",
            },
          ],
          "width": "100%",
        }
      }
    />
    <View
      collapsable={false}
      style={
        {
          "borderColor": "transparent",
          "borderLeftColor": "color-for-vote0",
          "borderRadius": 30,
          "borderWidth": 6,
          "height": "100%",
          "position": "absolute",
          "transform": [
            {
              "rotate": "40deg",
            },
          ],
          "width": "100%",
        }
      }
    />
    <View
      collapsable={false}
      style={
        {
          "borderColor": "transparent",
          "borderLeftColor": "color-for-vote0",
          "borderRadius": 30,
          "borderWidth": 6,
          "height": "100%",
          "position": "absolute",
          "transform": [
            {
              "rotate": "40deg",
            },
          ],
          "width": "100%",
        }
      }
    />
    <View
      collapsable={false}
      style={
        {
          "borderColor": "transparent",
          "borderLeftColor": "color-for-vote0",
          "borderRadius": 30,
          "borderWidth": 6,
          "height": "100%",
          "position": "absolute",
          "transform": [
            {
              "rotate": "40deg",
            },
          ],
          "width": "100%",
        }
      }
    />
    <View
      style={
        [
          {
            "backgroundColor": "#fff",
          },
          {
            "alignItems": "center",
            "borderRadius": 45,
            "height": 54,
            "justifyContent": "center",
            "width": 54,
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#11181C",
            },
            {
              "fontSize": 16,
            },
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            {
              "fontSize": 16,
              "fontWeight": "bold",
            },
          ]
        }
      >
        5
      </Text>
    </View>
  </View>
</View>
`;

exports[`VoteAverage renders correctly for voteAverage = 8 1`] = `
<View
  style={{}}
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "#D1D1D182",
          "borderRadius": 80,
          "borderWidth": 0,
          "height": 63,
          "justifyContent": "center",
          "position": "relative",
          "width": 63,
        },
        {
          "borderColor": "transparent",
        },
      ]
    }
  >
    <View
      collapsable={false}
      style={
        {
          "borderColor": "transparent",
          "borderLeftColor": "color-for-vote0",
          "borderRadius": 30,
          "borderWidth": 6,
          "height": "100%",
          "position": "absolute",
          "transform": [
            {
              "rotate": "130deg",
            },
          ],
          "width": "100%",
        }
      }
    />
    <View
      collapsable={false}
      style={
        {
          "borderColor": "transparent",
          "borderLeftColor": "color-for-vote0",
          "borderRadius": 30,
          "borderWidth": 6,
          "height": "100%",
          "position": "absolute",
          "transform": [
            {
              "rotate": "40deg",
            },
          ],
          "width": "100%",
        }
      }
    />
    <View
      collapsable={false}
      style={
        {
          "borderColor": "transparent",
          "borderLeftColor": "color-for-vote0",
          "borderRadius": 30,
          "borderWidth": 6,
          "height": "100%",
          "position": "absolute",
          "transform": [
            {
              "rotate": "40deg",
            },
          ],
          "width": "100%",
        }
      }
    />
    <View
      collapsable={false}
      style={
        {
          "borderColor": "transparent",
          "borderLeftColor": "color-for-vote0",
          "borderRadius": 30,
          "borderWidth": 6,
          "height": "100%",
          "position": "absolute",
          "transform": [
            {
              "rotate": "40deg",
            },
          ],
          "width": "100%",
        }
      }
    />
    <View
      style={
        [
          {
            "backgroundColor": "#fff",
          },
          {
            "alignItems": "center",
            "borderRadius": 45,
            "height": 54,
            "justifyContent": "center",
            "width": 54,
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#11181C",
            },
            {
              "fontSize": 16,
            },
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            {
              "fontSize": 16,
              "fontWeight": "bold",
            },
          ]
        }
      >
        8
      </Text>
    </View>
  </View>
</View>
`;

exports[`VoteAverage renders correctly for voteAverage = 10 1`] = `
<View
  style={{}}
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "#D1D1D182",
          "borderRadius": 80,
          "borderWidth": 0,
          "height": 63,
          "justifyContent": "center",
          "position": "relative",
          "width": 63,
        },
        {
          "borderColor": "transparent",
        },
      ]
    }
  >
    <View
      collapsable={false}
      style={
        {
          "borderColor": "transparent",
          "borderLeftColor": "color-for-vote0",
          "borderRadius": 30,
          "borderWidth": 6,
          "height": "100%",
          "position": "absolute",
          "transform": [
            {
              "rotate": "130deg",
            },
          ],
          "width": "100%",
        }
      }
    />
    <View
      collapsable={false}
      style={
        {
          "borderColor": "transparent",
          "borderLeftColor": "color-for-vote0",
          "borderRadius": 30,
          "borderWidth": 6,
          "height": "100%",
          "position": "absolute",
          "transform": [
            {
              "rotate": "40deg",
            },
          ],
          "width": "100%",
        }
      }
    />
    <View
      collapsable={false}
      style={
        {
          "borderColor": "transparent",
          "borderLeftColor": "color-for-vote0",
          "borderRadius": 30,
          "borderWidth": 6,
          "height": "100%",
          "position": "absolute",
          "transform": [
            {
              "rotate": "40deg",
            },
          ],
          "width": "100%",
        }
      }
    />
    <View
      collapsable={false}
      style={
        {
          "borderColor": "transparent",
          "borderLeftColor": "color-for-vote0",
          "borderRadius": 30,
          "borderWidth": 6,
          "height": "100%",
          "position": "absolute",
          "transform": [
            {
              "rotate": "40deg",
            },
          ],
          "width": "100%",
        }
      }
    />
    <View
      style={
        [
          {
            "backgroundColor": "#fff",
          },
          {
            "alignItems": "center",
            "borderRadius": 45,
            "height": 54,
            "justifyContent": "center",
            "width": 54,
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#11181C",
            },
            {
              "fontSize": 16,
            },
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            {
              "fontSize": 16,
              "fontWeight": "bold",
            },
          ]
        }
      >
        10
      </Text>
    </View>
  </View>
</View>
`;
